### PR TITLE
Update Singapore holidays: add 2023 Polling Day

### PR DESCRIPTION
**Pull request by mborsetti**
## Proposed change

Added Singapore Polling Day holiday on 1 September 2023, as Announced in state-associated press on 12 August 2023: https://www.straitstimes.com/singapore/politics/singapore-presidential-election-2023-polling-day-on-sept-1-nomination-day-on-aug-22.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] This PR is filed against `beta` branch of the repository
- [x] This PR doesn't contain any merge conflicts and has clean commit history
- [x] The code style looks good: `make pre-commit`
- [x] All tests pass locally: `make test`, `make tox` (we strongly encourage adding tests to your code)
- [x] The related [documentation][docs] has been added/updated (check off the box for free if no updates is required)

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Review by PPsyrius — COMMENTED** [review]
LGTM 🎉

File: tests/countries/test_singapore.py [hunk 1]
```diff
@@ -149,11 +149,12 @@ def test_2023(self):
         self.assertIn(date(2023, 6, 2), self.holidays)
         self.assertIn(date(2023, 6, 29), self.holidays)
         self.assertIn(date(2023, 8, 9), self.holidays)
+        self.assertIn(date(2023, 9, 1), self.holidays)
         self.assertIn(date(2023, 11, 12), self.holidays)
         self.assertIn(date(2023, 11, 13), self.holidays)
         self.assertIn(date(2023, 12, 25), self.holidays)
         # 2023: total holidays (11 + 3 falling on a Sunday)
```
Comment (arkid15r) [inline tests/countries/test_singapore.py:156]: It seems the comment needs to be updated as well.

**Review by arkid15r — COMMENTED** [review]
🇸🇬 LGTM

File: tests/countries/test_singapore.py (hunk 1 above)
Comment (mborsetti) [reply tests/countries/test_singapore.py]: Good catch! Done.

**Comment by arkid15r** [comment]
Thanks for keeping this up to date! Merged into beta.

**Pull request merged**